### DISCUSSION TITLE
Developing Web server

### DISCUSSION THREAD

**Discussion by ghazalialshafi** [VBAndCs/sVB-Small-Visual-Basic#33] [Ideas]
I want sVB to make a web server by which I can interact with HTML/JS in browser and send request to server and by this way utilize the ability to do Great task which Javascript can do and also interact with operating system api
The server will be hosting HTML/JS locally in Local Host and user will be interacting with it and it will be possible to us OS at same time and with the help of hostnames file custom domains which need to buy can be utilized to make a local website similar to what winginx do https://winginx.com/en

**Comment by VBAndCs** [comment]
@allcpu 
You can create extensions for sVB using VB.NEt or C#, where you can do whatever you want. The .NET framework contains all the necessry classses to deal with the web and Http protocols, and even parse and evaluate JS. It also contains a browser control that can be placed on the form and view web pages. But I don't advise to do any of that in sVB because it will be complicated due to its primitive compiler. The web page is a complex hierarchy of objects, while SB compiler doesn't recognize objects and sVB just give you the illusion of plain objects (only one level of members). 
This is where you need to move on to VB.NET or C#. SB (and sVB) are just a door to allow you enter the programming world.

Note:  WASM is replacing JS now, to allow you to use a high level programming language like C# to program the client-side, so I see no purpose in wasting the kids time in learning JS.
Thanks.
